packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -90,13 +90,40 @@ const getMediaGalleryEntries = (product, optionCodes, optionSelections) => {
     return value;
 };
 
+// We only want to display breadcrumbs for one category on a PDP even if a
+// product has multiple related categories. This function filters and selects
+// one category id for that purpose.
+const getBreadcrumbCategoryId = categories => {
+    const breadcrumbSet = new Set();
+    categories.forEach(({ breadcrumbs }) => {
+        // breadcrumbs can be `null`...
+        (breadcrumbs || []).forEach(({ category_id }) =>
+            breadcrumbSet.add(category_id)
+        );
+    });
+
+    // Filter out category data for categories that exist as breadcrumbs.
+    const leafCategories = categories.filter(
+        category => !breadcrumbSet.has(category.id)
```

Comment:
Given we only care about the first item returned, we should use `categories.find` so we can bail early.